velox/exec/TableScan.cpp

```diff
@@ -148,6 +148,13 @@ RowVectorPtr TableScan::getOutput() {
 
       if (connectorSplit->dataSource) {
         ++numPreloadedSplits_;
+        if (auto iter = std::find(
+                preloadingSplits_.begin(),
+                preloadingSplits_.end(),
+                connectorSplit);
+            iter != preloadingSplits_.end()) {
```

Comment:
This should always be true.  Maybe change it to a `VELOX_CHECK`

Reply:
Thank you for your reviewing @Yuhta . I observed in UT that when `maxDrivers` is greater than 1, `connectorSplit` may not always be found in `preloadingSplits_`. This inconsistency arises because the preloading split async source might be generated by one driver and consumed by another, then `connectorSplit` is tracked by a different instance of `preloadingSplits_`. While this logic isn't essential for resolving the memory leak problem, it's designed to expedite the removal of `connectorSplit` from `preloadingSplits_`. If this implementation proves confusing, I believe we can remove it. Please inform me of your thoughts on this matter.

Reply:
Updated, thanks.

Reply:
Where is this happening?  The async source should be generated and consumed by same driver.

Reply:
It's probably functionally working but not very clear code.  Since the splits can can be triggered to preload in one driver and processed in another, they should belong to task level.  How about we do this?
1. Move `preloadingSplits_` to `Task`
2. Add splits to it in `Task::getSplitLocked` after we call `preload`
3. Remove the one returned from `Task::getSplitLocked`
4. Close the remaining in `Task::terminate`

Reply:
Yes, make sense.

Reply:
Updated, thanks.